governance: add SIG governance and charter template (#150)
pingcap/community: governance/sig-governance.md

# SIG Governance

> This charter details on the governance guidelines of TiDB SIGs. It will be updated as needed to meet the current needs of the TiKV project.

Reviewer: ```suggestion
This charter details on the governance guidelines of TiDB SIGs. It will be updated as needed to meet the current needs of the TiDB Cimmunity.
```

## Organizational Governance

In order to standardize Special Interest Group efforts, create maximum transparency, and route contributors to the appropriate SIG, SIGs should follow these guidelines:

> - Create a charter and have it approved according to the SIG charter  process. Please refer [Create the SIG charter](#create-the-sig-charter) for more details.

Reviewer: ```suggestion
- Create a charter and have it approved according to the SIG charter process. Please refer [Create the SIG charter](#create-the-sig-charter) for more details.

```

> - Meet regularly, at least for 30 minutes every 2 weeks, except for long holidays

Reviewer: What if there is nothing to discuss, can they cancel the meeting?

> - Meet regularly, at least for 30 minutes every 2 weeks, except for long holidays

Reviewer 2: ```suggestion
- Meet regularly, at least for 30 minutes every 2 weeks, except for long holidays or occasions where all Tech Leads decides that there is nothing to discuss.
```

- Keep up-to-date meeting notes, linked from the SIG's page in the community repo
- Report activity in the weekly community meeting at least once every quarter
- Participate in release planning meetings and retrospectives, and burndown meetings, as needed
- Ensure related work happens in a project-owned github org and repository, with code and tests explicitly owned and supported by the SIG, including issue triage, PR reviews, test-failure response, bug  fixes, etc.
- The assignable tasks will be graded by difficulty levels and claimed by SIG members.
- Use the GitHub PR & issues, slack channels as the primary means of working, communicating, and collaborating.

## SIG Roles

> The SIG follows the role definitions in the [TiDB Developer Group](/architecture/README.md#tidb-developer-group) and [GOVERNANCE](/GOVERNANCE.md).

Reviewer: `GOVERNANCE.md` is missed.

Author: fixed

### SIG Members

- Inludes all Contributors, Active Contributors, Reviewers, and Committers under the SIG
- SHOULD maintain health of the SIG
- SHOULD show sustained contributions to the SIG
- SHOULD hold corresponding responsibilities in the SIG as documented in [TiDB Developer Group](/architecture/README.md#tidb-developer-group)
- MAY participate in decision making of the SIG

> - Members are tracked in `membership.md` in the specific SIG (Contributors are listed in the aggregated contributor list for maintainence convenience)

Reviewer: actually, we are using `member-list.md`

### Tech Lead

  - **Number**: 2-3. Initially assigned by Maintainers
  - **Requirement**:
    - MUST be a committer from the specific SIG

>     - MUST have demonstrated deep understanding of the SIG project

Reviewer: ```suggestion
    - MUST have demonstrated deep understanding of the SIG
```

  - **Term**: 1 year
  - **Election:**
      - Nominated by the Tech Leads or self-nominate
      - Gained supermajority votes of Maintainers

>       - Done through PR to update the `membership.md` file in the SIG

Reviewer: ```suggestion
      - Done through PR to update the `member-list.md` file in the SIG
```

- **Responsibilities**
    - Creating new projects under the SIG
    - Triaging issues/tasks in the SIG
    - Mentoring members of the SIG
    - Organizing discussions and drawing conclusions for decision making within the SIG
    - Ensuring health and sustained development of the SIG and grwoth of SIG members

>     - Making roadmap for the SIG

Reviewer: maybe we can remove it for now. currently, we only have a roadmap for the whole tidb product.

## Member Promotion Mechanism

Members of the SIG can be promoted to a higher role based on the rules defined in [TiDB Developer Group](/architecture/README.md#tidb-developer-group). 

## Member Exit Mechanism

There are two exit situations:

- Members MAY decide to step down at anytime for personal reasons by opening the PR to modify the `member-list.md`in the Specific SIG. The PR can be merged once it gets two approvals from the higher role.

- Members of a role SHOULD be removed from the SIG if they have not communicated a leave of absence and either cannot be reached for more than 1 month or are not fulfilling their documented responsibilities for more than 1 month. Tech Leads will assess the communication and contribution status monthly to update the membership list. Each change (removal), depending on the role, requires the following:

    - For Committer or Tech Leads, a supermajority votes from Maintainers are required
    - For Reviewers or lower roles, two approvals from the immediate higher role is required

Once the change is approved, the corresponding profile will be moved to the emeritus role list.

## SIG Lifecycle

### Creation

> 1. Copy the [SIG Charter Template](SIG-CHARTER-TEMPLATE.md).

Reviewer: how about to create a TEMPLATE folder in `special-interest-groups`, then put all the template files belonging to a SIG to that folder?

so that SIG creators only need to copy that template folder，modify file content in the copied folder.

Author: Good idea, I'll do it later.

2. Modify the content in the template that needs to be defined for the specific SIG

> 3. Initiate a PR in [special-interest-groups](https://github.com/pingcap/community/tree/master/special-interest-groups) with the SIG Charter, and propose the SIG Charter. This must be approved by the supermajority of the PMC members.

Reviewer: ```suggestion
3. Initiate a PR in [special-interest-groups](/special-interest-groups) with the SIG Charter, and propose the SIG Charter. This must be approved by the supermajority of the PMC members.
```

Reviewer: seems we need to create a PMC team in the `pingcap` org.

> 4. Create a public and private Slack Channel in tikv-wg.slack.com to discuss the SIG operation related matters.

Reviewer: ```suggestion
4. Create a public Slack Channel in [tidbcommunity](https://pingcap.com/tidbslack/) to discuss the SIG operation related matters. Channel name is recommended to be prefixed with `sig-`
```

> 5. Create a Zoom room for regular meetings, as well as other online discussions.

Reviewer: ```suggestion
5. Create a public [Zoom](https://zoom.com/) url for regular meetings, as well as other online discussions.
```

> 6. Announce the establishment of a new SIG in the TiKV community.

Reviewer: SIG creator also needs to create a public google doc to summary the regular SIG work report and record the regular meeting notes.

Author: Added

- Create a charter and have it approved according to the SIG charter process. Please refer to [Create the SIG charter](#create-the-sig-charter) for more details.

- Meet regularly, at least for 30 minutes every 2 weeks, except for long holidays or occasions where all Tech Leads decide that there is nothing to discuss.
- Keep up-to-date meeting notes, linked from the SIG's page in the community repo
- Report activity in the weekly community meeting at least once every quarter
- Participate in release planning meetings and retrospectives, and burndown meetings, as needed
- Ensure related work happens in a project-owned github org and repository, with code and tests explicitly owned and supported by the SIG, including issue triage, PR reviews, test-failure response, bug  fixes, etc.
- The assignable tasks will be graded by difficulty levels and claimed by SIG members.
- Use the GitHub PR & issues, slack channels as the primary means of working, communicating, and collaborating.

## SIG Roles

The SIG follows the role definitions in the [TiDB Developer Group](/architecture/README.md#tidb-developer-group) and [GOVERNANCE](/GOVERNANCE.md).

### SIG Members

> - Inludes all Contributors, Active Contributors, Reviewers, and Committers under the SIG

Author: ```suggestion
- Include all Contributors, Active Contributors, Reviewers, and Committers under the SIG
```

# SIG Governance

> Special Interest Groups (SIGs) are persistent open groups that focus on a module of TiDB. SIGs must have open and transparent proceedings. The purpose of a SIG is to own and develop a module of TiDB. This document details on the governance guidelines of TiDB SIGs. It will be updated as needed to meet the current needs of the TiDB Community.

Reviewer 3: ```suggestion
Special Interest Groups (SIGs) are persistent open groups that focus on some modules of TiDB. SIGs must have open and transparent proceedings. The purpose of a SIG is to own and develop a module of TiDB. This document details on the governance guidelines of TiDB SIGs. It will be updated as needed to meet the current needs of the TiDB Community.
```

Reviewer 3: I think the definition of sig in K8s community is more proper:

```
The Kubernetes project is organized primarily into Special Interest Groups, or SIGs. Each SIG is comprised of members from multiple companies and organizations, with a common purpose of advancing the project with respect to a specific topic, such as Networking or Documentation. Our goal is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting work done, making decisions, and onboarding new contributors. Every identifiable subpart of the project (e.g., github org, repository, subdirectory, API, test, issue, PR) is intended to be owned by some SIG.

Areas covered by SIGs may be vertically focused on particular components or functions, cross-cutting/horizontal, spanning many/all functional areas of the project, or in support of the project itself. Examples:

Vertical: Network, Storage, Node, Scheduling
Horizontal: Scalability, Architecture
Project: Testing, Release, Docs, PM, Contributor Experience
```

> Special Interest Groups (SIGs) are persistent open groups that focus on a module of TiDB. SIGs must have open and transparent proceedings. The purpose of a SIG is to own and develop a module of TiDB. This document details on the governance guidelines of TiDB SIGs. It will be updated as needed to meet the current needs of the TiDB Community.

Reviewer 3: I think the definitions of SIGs in K8s community is proper:

```
The Kubernetes project is organized primarily into Special Interest Groups, or SIGs. Each SIG is comprised of members from multiple companies and organizations, with a common purpose of advancing the project with respect to a specific topic, such as Networking or Documentation. Our goal is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting work done, making decisions, and onboarding new contributors. Every identifiable subpart of the project (e.g., github org, repository, subdirectory, API, test, issue, PR) is intended to be owned by some SIG.

Areas covered by SIGs may be vertically focused on particular components or functions, cross-cutting/horizontal, spanning many/all functional areas of the project, or in support of the project itself. Examples:

Vertical: Network, Storage, Node, Scheduling
Horizontal: Scalability, Architecture
Project: Testing, Release, Docs, PM, Contributor Experience
```
A sig can cover more than one modules. For example, TiDB community can have sig-web, sig-usability, sig-ui, etc.

## Organizational Governance

In order to standardize Special Interest Group efforts, create maximum transparency, and route contributors to the appropriate SIG, SIGs should follow these guidelines:

- Create a `sig-${name}` folder in [special interest group directory](../special-interest-groups) and have it approved according to the SIG charter process. Please refer to [Creation](#Creation) for more details.
- Meet regularly, at least for 30 minutes every 2 weeks, except for long holidays or occasions where all Tech Leads decide that there is nothing to discuss.
- Keep up-to-date meeting notes, linked from the SIG's page in the community repo
- Report activity in the weekly community meeting at least once every quarter
- Participate in release planning meetings and retrospectives, and burndown meetings, as needed
- Ensure related work happens in a project-owned GitHub organization and repository, with code and tests explicitly owned and supported by the SIG, including issue triage, PR reviews, test-failure response, bug fixes, etc.
- The assignable tasks will be graded by difficulty levels and claimed by SIG members.

> - Use the GitHub PR & issues, Slack channels as the primary means of working, communicating, and collaborating.

Reviewer 3: ```suggestion
```

Reviewer 3: It's better not to define this.

# SIG Governance

Special Interest Groups (SIGs) are persistent open groups that focus on a module of TiDB. SIGs must have open and transparent proceedings. The purpose of a SIG is to own and develop a module of TiDB. This document details on the governance guidelines of TiDB SIGs. It will be updated as needed to meet the current needs of the TiDB Cimmunity.

## Organizational Governance

In order to standardize Special Interest Group efforts, create maximum transparency, and route contributors to the appropriate SIG, SIGs should follow these guidelines:

> - Create a `sig-${name}` folder in [special interest group directory](../special-interest-groups) and have it approved according to the SIG charter process. Please refer to [Creation](#Creation) for more details.

Reviewer 2: I think it is clear in the Creation section.

- Meet regularly, at least for 30 minutes every 2 weeks, except for long holidays or occasions where all Tech Leads decide that there is nothing to discuss.
- Keep up-to-date meeting notes, linked from the SIG's page in the community repo
- Report activity in the weekly community meeting at least once every quarter
- Participate in release planning meetings and retrospectives, and burndown meetings, as needed
- Ensure related work happens in a project-owned github org and repository, with code and tests explicitly owned and supported by the SIG, including issue triage, PR reviews, test-failure response, bug  fixes, etc.
- The assignable tasks will be graded by difficulty levels and claimed by SIG members.

> - Use the GitHub PR & issues, slack channels as the primary means of working, communicating, and collaborating.

Reviewer 2: ```suggestion
- Use the GitHub PR & issues, Slack channels as the primary means of working, communicating, and collaborating.
```

## SIG Roles

The SIG follows the role definitions in the [TiDB Developer Group](/architecture/README.md#tidb-developer-group) and [GOVERNANCE](/GOVERNANCE.md).

### SIG Members

- Include all Contributors, Active Contributors, Reviewers, and Committers under the SIG
- SHOULD maintain health of the SIG
- SHOULD show sustained contributions to the SIG
- SHOULD hold corresponding responsibilities in the SIG as documented in [TiDB Developer Group](/architecture/README.md#tidb-developer-group)
- MAY participate in decision making of the SIG

> - Members are tracked in `membership.md` in the specific SIG (Contributors are listed in the aggregated contributor list for maintainence convenience)

Reviewer 2: ```suggestion
- Members are tracked in ``member-list.md` in the specific SIG
```

## SIG Lifecycle

### Creation

* Before creating a special interest group, make sure you get the following ready:
  * Ensure all SIG Tech Leads, and other SIG roles are community members
  * Have the answers to following questions:
    * What is the exact module this group focused on?
    * What is the artifact that this group will deliver, and to whom?
    * What are the meeting mechanics (frequency, duration, roles)?

* Once you have met the above criteria, you can create the SIG according to the  following steps:

	1. Create a `sig-${name}` folder in [special interest group directory](../special-interest-groups) (modify `${name}` to your Special Interest Group short name) according to [SIG Template](../special-interest-groups/sig-template), and modify the **Active Special Interest Groups** list in the [Special Interest Groups](../special-interest-groups/README.md) page. 
	2. Create a public Slack Channel in [tidbcommunity](https://pingcap.com/tidbslack/) to discuss the SIG operation related matters. Channel name is recommended to be prefixed with `sig-${name}`
	3. Create a public [Zoom](https://zoom.com/) url for regular meetings, as well as other online discussions.

> 	4. Create a public Google Doc to summary the regular SIG work report and record the regular meeting notes.

Reviewer 2: ```suggestion
	4. Create a public Google Doc to summarize the regular SIG work report and record the regular meeting notes.
```

# SIG Governance

The TiDB project is organized primarily into Special Interest Groups, or SIGs. Each SIG is comprised of members from multiple companies and organizations, with **a common purpose of advancing the project with respect to a specific topic, such as Networking or Documentation**. Our goal is to enable a distributed decision structure and code ownership, as well as providing focused instructions for getting work done, making decisions, and onboarding new contributors. Every identifiable subpart of the project (e.g., github org, repository, subdirectory, API, test, issue, PR) is intended to be owned by some SIG.

Areas covered by SIGs may be vertically focused on particular components or functions, cross-cutting/horizontal, spanning many/all functional areas of the project, or in support of the project itself. Examples:

- Vertical: Network, Storage, Scheduling
- Horizontal: Scalability, Architecture
- Project: Testing, Docs,Contributor Experience

> 

Reviewer 2: Do we have these setups or just copying them from K8s?

In order to standardize Special Interest Group efforts, create maximum transparency, and route contributors to the appropriate SIG, SIGs should follow these guidelines:

> - Create a charter and have it approved according to the SIG charter process. Please refer [Create the SIG charter](#create-the-sig-charter) for more details.

Reviewer 2: ```suggestion
- Create a charter and have it approved according to the SIG charter process. Please refer to [Create the SIG charter](#create-the-sig-charter) for more details.
```
